lib/net/ber.rb

```diff
@@ -296,9 +296,8 @@ def to_arr
 class Net::BER::BerIdentifiedString < String
   attr_accessor :ber_identifier
   def initialize args
+    args.force_encoding('UTF-8') if args.respond_to(:force_encoding)
     super args
-    # LDAP uses UTF-8 encoded strings
-    self.encode('UTF-8') if self.respond_to?(:encoding) rescue self
```

Comment:
This line is effectively a noop. `self.encode` returns a new string and doesn't modify itself. To modify itself the call should have been `self.encode!`.